tests/test_client_device.py

```diff
@@ -1354,14 +1462,51 @@ def on_mdib_id_change(flag: bool):
         self.assertEqual(len(cl_mdib.states.objects), len(self.sdc_device.mdib.states.objects))
         self.assertEqual(len(cl_mdib.context_states.objects), len(self.sdc_device.mdib.context_states.objects))
 
-
-class Test_DeviceCommonHttpServer(unittest.TestCase):
-
+    def test_descriptor_lookup(self):
+        """Verify that descriptor lookup works on client mdib."""
+        mdib = ConsumerMdib(self.sdc_client)
+        mdib.init_mdib()
+
+        self.assertEqual(104, len(mdib.descriptions.objects))
+        self.assertEqual(104, len(mdib.descriptions.handle))
+        self.assertEqual(20, len(mdib.descriptions.parent_handle))
+        self.assertEqual(18, len(mdib.descriptions.NODETYPE))
+        self.assertEqual(65, len(mdib.descriptions.coding))
+
+    def test_descriptor_lookup_condition_signaled(self):
+        """Verify that condition signaled descriptor lookup works on client mdib."""
+        mdib = ConsumerMdib(self.sdc_client)
+        mdib.init_mdib()
+        self.assertEqual(1, len(mdib.descriptions.condition_signaled))
+        descriptors = mdib.descriptions.condition_signaled.get('0xD3C00100')
+        self.assertEqual(1, len(descriptors))
+        self.assertEqual('0xD3C00100.loc.Vis', descriptors[0].Handle)
+
+    def test_descriptor_lookup_source(self):
+        """Verify that source descriptor lookup works on client mdib."""
+        mdib = ConsumerMdib(self.sdc_client)
+        mdib.init_mdib()
+        self.assertEqual(len(mdib.descriptions.source), 4)
+        descriptors = mdib.descriptions.source.get('3569')
+        self.assertEqual(1, len(descriptors))
+        self.assertEqual('0xD3C00100', descriptors[0].Handle)
+        descriptors = mdib.descriptions.source.get('0x34F00150')

```

Comment:
maybe add following Source
<pm:Source>0x34F00150</pm:Source>

 to AlertCondition with Handle="0xD3C00108" in MDIB

and add following:

```Python
        descriptors = mdib.descriptions.source.get('0x34F00150')
        self.assertEqual(2, len(descriptors))
        exp_handles = [d.Handle for d in descriptors]
        self.assertIn('0xD3C00109', exp_handles)
        self.assertIn('0xD3C00108', exp_handles)

        with self.assertRaises(ValueError) as cm:
            mdib.descriptions.source.get_one('0x34F00150')
        self.assertIn('has 2 objects', str(cm.exception))
```